pkg/asset/installconfig/azure/validation.go

```diff
@@ -41,6 +42,13 @@ func Validate(client API, ic *types.InstallConfig) error {
 	allErrs = append(allErrs, validateNetworks(client, ic.Azure, ic.Networking.MachineNetwork, field.NewPath("platform").Child("azure"))...)
 	allErrs = append(allErrs, validateRegion(client, field.NewPath("platform").Child("azure").Child("region"), ic.Azure)...)
 	allErrs = append(allErrs, validateInstanceTypes(client, ic)...)
+	if ic.Azure.CloudName == aztypes.StackCloud {
```

Comment:
either
```suggestion
	if ic.Azure.CloudName == aztypes.StackCloud && ic.Azure.ClusterOSImage != "" {
```

or just

```suggestion
	if ic.Azure.ClusterOSImage != "" {
```

The current scenario will cause failures in UPI or not the best error message on IPI:

```shell
# yq e .platform.azure c/install-config.yaml 
armEndpoint: https://management.ppe3.stackpoc.com
baseDomainResourceGroupName: openshiftInstallerRG
cloudName: AzureStackCloud
region: ppe3

# ./openshift-install create manifests --dir c
INFO Credentials loaded from file "/root/.azure/osServicePrincipal.json" 
FATAL failed to fetch Master Machines: failed to load asset "Install Config": platform.azure.clusterOSImage: Invalid value: "": clusterOSImage must be in the Azure Stack environment 
```